packages/contracts/src/token/ERC1155/MultiTokenVault.sol

```diff
@@ -300,7 +300,7 @@ abstract contract MultiTokenVault is
         uint256 assets,
         uint256 shares
     ) internal virtual nonReentrant {
-        if (caller != owner && isApprovedForAll(owner, caller)) {
+        if (caller != owner && !isApprovedForAll(owner, caller)) {
             revert MultiTokenVault__CallerMissingApprovalForAll(caller, owner);
         }
```

Comment:
please make sure there are explicit test case for all scenarios:
- owner - allowed
- approvedForAll - allowed
- not owner and not approved for all - fails